### PR TITLE
[Core] add input tolerance in search is inside

### PR DESCRIPTION
**Pull request by vdivine**
Tolerance is a constexpr as 1.0E-12. For some cases, i need to use a smaller tolerance, otherwise a given point is not found anywhere.
This PR allows user to input a tolerance value.

### CODE REVIEW

**Comment by vdivine** [comment]
ping @ddiezrod @pooyan-dadvand @loumalouomega

**Comment by loumalouomega** [comment]
I would set it on the constructor, opinion @pooyan-dadvand ?

**Review by rubenzorrilla — CHANGES_REQUESTED** [review]
This breaks current API. I'd follow @loumalouomega's suggestion instead.

**Comment by vdivine** [comment]
ping @loumalouomega @ddiezrod

File: kratos/python/add_search_strategies_to_python.cpp [hunk 1]
```diff
@@ -503,8 +503,8 @@ void AddSearchStrategiesToPython(pybind11::module& m)
     using ConditionsContainerType = ModelPart::ConditionsContainerType;
 
     py::class_<GeometricalObjectsBins, GeometricalObjectsBins::Pointer>(m, "GeometricalObjectsBins")
-    .def(py::init<ElementsContainerType&>())
-    .def(py::init<ConditionsContainerType&>())
+    .def(py::init<ElementsContainerType&, double>())
```
Comment (ddiezrod) [inline kratos/python/add_search_strategies_to_python.cpp:508]: I'd add  the constructor without the tolerance too

**Comment by ddiezrod** [comment]
OK for me, I leave the approval for @rubenzorrilla

**Comment by vdivine** [comment]
ping @rubenzorrilla

File: kratos/spatial_containers/geometrical_objects_bins.h [hunk 2]
```diff
@@ -90,8 +92,8 @@ class KRATOS_API(KRATOS_CORE) GeometricalObjectsBins
      * @tparam TContainer The container type
      */
     template<typename TContainer>
-    GeometricalObjectsBins(TContainer& rGeometricalObjectsVector)
-        : GeometricalObjectsBins(rGeometricalObjectsVector.begin(), rGeometricalObjectsVector.end())
+    GeometricalObjectsBins(TContainer& rGeometricalObjectsVector, double Tolerance = 1e-12)
```
Comment (rubenzorrilla) [inline kratos/spatial_containers/geometrical_objects_bins.h]:
```suggestion
    GeometricalObjectsBins(
        TContainer& rGeometricalObjectsVector,
        const double Tolerance = 1e-12)
```

File: kratos/spatial_containers/geometrical_objects_bins.h [hunk 3]
```diff
@@ -68,17 +68,19 @@ class KRATOS_API(KRATOS_CORE) GeometricalObjectsBins
     template<typename TIteratorType>
     GeometricalObjectsBins(
         TIteratorType GeometricalObjectsBegin,
-        TIteratorType GeometricalObjectsEnd
+        TIteratorType GeometricalObjectsEnd,
+        double Tolerance = 1e-12
```
Comment (rubenzorrilla) [inline kratos/spatial_containers/geometrical_objects_bins.h]:
```suggestion
        const double Tolerance = 1e-12
```

**Review by rubenzorrilla — APPROVED** [review]
Should be OK now. Thanks.

**Pull request merged**